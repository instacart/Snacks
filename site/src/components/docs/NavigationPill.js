import React, {Fragment} from 'react'
import Playground from '../Playground'

export default () => (

NavigationPill example:

<Playground>
  {`
    <div>
      <NavigationPill
        isActive={false}
        onClick={e => {
          e.preventDefault()
          console.log('NavigationPill click!', "bananas")
        }}
        text={"bananas"}
      />
      <NavigationPill
        isActive={true}
        onClick={e => {
          e.preventDefault()
          console.log('NavigationPill click!', "carrots")
        }}
        text={"carrots"}
      />
      <NavigationPill
        isActive={false}
        onClick={e => {
          e.preventDefault()
          console.log('NavigationPill click!', "watermelon")
        }}
        text={"watermelon"}
      />
    </div>
`}
</Playground>

)
