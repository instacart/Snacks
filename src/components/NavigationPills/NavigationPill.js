import React              from 'react'
import PropTypes          from 'prop-types'
import Radium             from 'radium'
import colors             from '../../styles/colors'
import withTheme          from '../../styles/themer/withTheme'
import { themePropTypes } from '../../styles/themer/utils'

const styles = {
  container: {
    display: 'inline-block'
  },
  main: {
    default: {
      padding: '12px 16px',
      display: 'block',
      fontSize: '14px',
      backgroundColor: colors.GRAY_97,
      borderRadius: '24px',
      margin: '0 4px',
      lineHeight: '1.2',
      transition: 'background-color 150ms ease-in-out',

      ':hover': {
        textDecoration: 'none',
        backgroundColor: colors.GRAY_93
      },

      ':focus': {
        textDecoration: 'none',
        backgroundColor: colors.GRAY_93,
        outline: 'none'
      }
    }
  }
}

const NavigationPill = props => {
  const { isActive, snacksTheme, text } = props
  const { primaryForeground } = snacksTheme.colors

  const activeStyles = {
    backgroundColor: primaryForeground,
    color: colors.WHITE,

    ':hover': {
      backgroundColor: primaryForeground
    },
    ':focus': {
      backgroundColor: primaryForeground,
    }
  }

  return (
    <li style={styles.container} role='tab'>
      <a
        href={props.path || '#'}
        data-bypass={true}
        onClick={e => props.onClick(e, props)}
        style={[
          styles.main.default,
          { color: primaryForeground },
          isActive && activeStyles
        ]}
        key={`pill-anchor-${text}`}
      >
        {text}
      </a>
    </li>
  )
}

NavigationPill.propTypes = {
  /** determines wether or not active styles are applied */
  isActive: PropTypes.bool,
  /** Callback function called after pill click
   * @param {SyntheticEvent} event The react `SyntheticEvent`
   * @param {props} object All the props passed to the component
  */
  onClick: PropTypes.func,
  /** url used for href property */
  path: PropTypes.string,
  /** snacks theme attributes */
  snacksTheme: themePropTypes,
  /** text to appear inside pill */
  text: PropTypes.string
}

NavigationPill.defaultProps = {
  isActive: false
}

export default withTheme(Radium(NavigationPill))
