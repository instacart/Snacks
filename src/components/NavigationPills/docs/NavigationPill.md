NavigationPill example:
    <div>
      <NavigationPill
        isActive={false}
        onClick={e => {
          e.preventDefault()
          console.log('NavigationPill click!', "bananas")
        }}
        text={"bananas"}
      />
      <NavigationPill
        isActive={true}
        onClick={e => {
          e.preventDefault()
          console.log('NavigationPill click!', "carrots")
        }}
        text={"carrots"}
      />
      <NavigationPill
        isActive={false}
        onClick={e => {
          e.preventDefault()
          console.log('NavigationPill click!', "watermelon")
        }}
        text={"watermelon"}
      />
    </div>
