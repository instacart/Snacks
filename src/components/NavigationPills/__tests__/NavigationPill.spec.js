import React from 'react'
import renderer from 'react-test-renderer'
import NavigationPill from '../NavigationPill'

it('renders active pill correctly', () => {
  const tree = renderer.create(
    <NavigationPill
      isActive={true}
      text={'Active pill'}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders inactive pill correctly', () => {
  const tree = renderer.create(
    <NavigationPill
      isActive={false}
      text={'Not active pill'}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders href pill correctly', () => {
  const tree = renderer.create(
    <NavigationPill
      isActive={false}
      text={'Pill with href'}
      path={'/store/doms-cool-store'}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('renders element attributes on pill correctly', () => {
  const tree = renderer.create(
    <NavigationPill
      text={'pill with attributes'}
      elementAttributes={{ ariaLabel:'this is an aria label', role:'tab'}}
    />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})
