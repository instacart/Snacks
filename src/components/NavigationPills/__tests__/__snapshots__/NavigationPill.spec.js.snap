// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders active pill correctly 1`] = `
<li
  data-radium={true}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <a
    data-bypass={true}
    data-radium={true}
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "#43B02A",
        "borderRadius": "24px",
        "color": "#FFFFFF",
        "display": "block",
        "fontSize": "14px",
        "lineHeight": "1.2",
        "margin": "0 4px",
        "padding": "12px 16px",
        "transition": "background-color 150ms ease-in-out",
      }
    }
  >
    Active pill
  </a>
</li>
`;

exports[`renders element attributes on pill correctly 1`] = `
<li
  ariaLabel="this is an aria label"
  data-radium={true}
  role="tab"
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <a
    data-bypass={true}
    data-radium={true}
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "#F7F7F7",
        "borderRadius": "24px",
        "color": "#43B02A",
        "display": "block",
        "fontSize": "14px",
        "lineHeight": "1.2",
        "margin": "0 4px",
        "padding": "12px 16px",
        "transition": "background-color 150ms ease-in-out",
      }
    }
  >
    pill with attributes
  </a>
</li>
`;

exports[`renders href pill correctly 1`] = `
<li
  data-radium={true}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <a
    data-bypass={true}
    data-radium={true}
    href="/store/doms-cool-store"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "#F7F7F7",
        "borderRadius": "24px",
        "color": "#43B02A",
        "display": "block",
        "fontSize": "14px",
        "lineHeight": "1.2",
        "margin": "0 4px",
        "padding": "12px 16px",
        "transition": "background-color 150ms ease-in-out",
      }
    }
  >
    Pill with href
  </a>
</li>
`;

exports[`renders inactive pill correctly 1`] = `
<li
  data-radium={true}
  style={
    Object {
      "display": "inline-block",
    }
  }
>
  <a
    data-bypass={true}
    data-radium={true}
    href="#"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    style={
      Object {
        "backgroundColor": "#F7F7F7",
        "borderRadius": "24px",
        "color": "#43B02A",
        "display": "block",
        "fontSize": "14px",
        "lineHeight": "1.2",
        "margin": "0 4px",
        "padding": "12px 16px",
        "transition": "background-color 150ms ease-in-out",
      }
    }
  >
    Not active pill
  </a>
</li>
`;
